### ISSUE TITLE
InvalidArgument: 'expects a public static callable, but a non-static callable provided'

### ISSUE THREAD

**Issue by loevgaard** [vimeo/psalm#10936]
Between version 5.23.1 and 5.24.0 there is a difference in Psalm handling this code:

```php
<?php

declare(strict_types=1);

namespace App\Twig\Extension;

use Twig\Extension\AbstractExtension;
use Twig\TwigFunction;

final class Extension extends AbstractExtension
{
    public function getFunctions(): array
    {
        return [
            new TwigFunction('previous_checkout_step', [Runtime::class, 'previousCheckoutStep']),
            new TwigFunction('next_checkout_step', [Runtime::class, 'nextCheckoutStep']),
            new TwigFunction('taxon', [Runtime::class, 'taxon']),
        ];
    }
}
```

In `5.24.0` Psalm gives me these errors:

```text
ERROR: InvalidArgument - src/Twig/Extension/Extension.php:15:56 - Argument 2 of Twig\TwigFunction::__construct expects a public static callable, but a non-static callable provided (see https://psalm.dev/004)
            new TwigFunction('previous_checkout_step', [Runtime::class, 'previousCheckoutStep']),


ERROR: InvalidArgument - src/Twig/Extension/Extension.php:16:52 - Argument 2 of Twig\TwigFunction::__construct expects a public static callable, but a non-static callable provided (see https://psalm.dev/004)
            new TwigFunction('next_checkout_step', [Runtime::class, 'nextCheckoutStep']),


ERROR: InvalidArgument - src/Twig/Extension/Extension.php:17:39 - Argument 2 of Twig\TwigFunction::__construct expects a public static callable, but a non-static callable provided (see https://psalm.dev/004)
            new TwigFunction('taxon', [Runtime::class, 'taxon']),
```

In `5.23.1` I get no errors.

The `constructor` for `TwigFunction` looks like this:

```php
/**
 * @param callable|array{class-string, string}|null $callable A callable implementing the function. If null, you need to overwrite the "node_class" option to customize compilation.
 */
public function __construct(string $name, $callable = null, array $options = [])
{
    $this->name = $name;
    $this->callable = $callable;
    $this->options = array_merge([
        'needs_environment' => false,
        'needs_context' => false,
        'is_variadic' => false,
        'is_safe' => null,
        'is_safe_callback' => null,
        'node_class' => FunctionExpression::class,
        'deprecated' => false,
        'alternative' => null,
    ], $options);
}
```

I am not able to reproduce the bug unfortunately: https://psalm.dev/.

I have no idea, but somehow I think it's related to this PR: https://github.com/vimeo/psalm/pull/10839

**Comment by VincentLanglet** [comment]
A PR is opened https://github.com/vimeo/psalm/pull/10935

**Comment by weirdan** [comment]
Fixed in #10935

**Comment by loevgaard** [comment]
I think this issue was reintroduced in 5.25.0. I am getting the same error nonetheless
